projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx

```diff
@@ -41,95 +42,71 @@ const StandaloneMode = () => {
 
-	useEffect( () => {
+	const handleShareDebugDataChange = useCallback( () => {
 		setSettings( {
-			jetpack_waf_share_debug_data: jetpackWafShareDebugData,
+			...settings,
+			jetpack_waf_share_debug_data: ! settings.jetpack_waf_share_debug_data,
 		} );
-	}, [ jetpackWafShareDebugData ] );
-
-	return (
-		<div className={ styles[ 'share-data-section' ] }>
-			<Title mb={ 2 }>{ __( 'Share detailed data with Jetpack', 'jetpack-protect' ) }</Title>
-			<div className={ styles[ 'footer-checkbox' ] }>
-				<CheckboxControl
-					checked={ Boolean( settings.jetpack_waf_share_debug_data ) }
-					onChange={ handleShareDebugDataChange }
-					disabled={ isUpdating }
-					__nextHasNoMarginBottom={ true }
-				/>
-				<Text>
-					{ __(
-						'Allow Jetpack to collect detailed data from blocked requests to enhance firewall protection and accuracy.',
-						'jetpack-protect'
-					) }
-				</Text>
-			</div>
-		</div>
-	);
-};
-
-const ShareData = () => {
-	const { config, isUpdating, toggleShareData } = useWafData();
-	const { jetpackWafShareData } = config || {};
-	const { showSuccessNotice, showErrorNotice } = useNotices();
-
-	const [ settings, setSettings ] = useState( {
-		jetpack_waf_share_data: jetpackWafShareData,
-	} );
-
-	const handleShareDataChange = useCallback( () => {
-		setSettings( { ...settings, jetpack_waf_share_data: ! settings.jetpack_waf_share_data } );
-		toggleShareData()
+		toggleShareDebugData()
 			.then( () => showSuccessNotice( __( 'Changes saved.', 'jetpack-protect' ) ) )
 			.catch( () => {
 				showErrorNotice();
 			} );
-	}, [ settings, toggleShareData, showSuccessNotice, showErrorNotice ] );
+	}, [ settings, toggleShareDebugData, showSuccessNotice, showErrorNotice ] );
 
 	useEffect( () => {
 		setSettings( {
 			jetpack_waf_share_data: jetpackWafShareData,
+			jetpack_waf_share_debug_data: jetpackWafShareDebugData,
 		} );
-	}, [ jetpackWafShareData ] );
+	}, [ jetpackWafShareData, jetpackWafShareDebugData ] );
 
 	return (
-		<div className={ styles[ 'share-data-section' ] }>
-			<Title mb={ 2 }>{ __( 'Share basic data with Jetpack', 'jetpack-protect' ) }</Title>
-			<div className={ styles[ 'footer-checkbox' ] }>
-				<CheckboxControl
-					checked={ Boolean( settings.jetpack_waf_share_data ) }
-					onChange={ handleShareDataChange }
-					disabled={ isUpdating }
-					__nextHasNoMarginBottom={ true }
-				/>
-				<Text>
-					{ __(
-						'Allow Jetpack to collect basic data from blocked requests to improve firewall protection and accuracy.',
-						'jetpack-protect'
-					) }
-				</Text>
-			</div>
+		<div className={ styles[ 'share-data' ] }>
+			<Title mb={ 0 }>{ __( 'Share data with Jetpack', 'jetpack-protect' ) }</Title>
+			<ToggleControl
+				className={ styles[ 'share-data-toggle' ] }
+				checked={ Boolean( settings.jetpack_waf_share_data ) }
+				onChange={ handleShareDataChange }
+				disabled={ isUpdating }
+				size="small"
+				label={ __( 'Share basic data with Jetpack', 'jetpack-protect' ) }
```

Comment:
```suggestion
				label={ __( 'Share basic data', 'jetpack-protect' ) }
```
Now that we have a general title for both toggles, I suppose we can simplify the `label` text. Maybe even further to `Basic data` and `Detailed data`? Out of scope for this task, but I also wonder if the `help` copy could better describe the toggle functionality, we're basically just repeating the same thing for each with basic/detailed.